ansys/mapdl/core/logging.py

```diff
@@ -0,0 +1,499 @@
+    >>> LOG.debug('This is LOG debug message.')
+
+       | Level    | Instance        | Module           | Function             | Message
+    |----------|-----------------|------------------|----------------------|--------------------------------------------------------
+    | DEBUG    |                 |  __init__        | <module>             | This is LOG debug message.
+
+
+By default, ``_Global_`` does not output to file or stdout. You should activate it from ``__init__``.
+
+
+### Instance logger
+Every time that the class ``_MapdlCore`` is instantiated, a logger is created and stored in two places:
+
+* ``_MapdlCore._log``. For backward compatibility.
+* ``LOG._instances``. This field is a ``dict`` where the key is the name of the created logger.
+
+These instance loggers inheritate the ``_Global_`` output handlers and logging level unless otherwise specified.
+
+You can use this logger like this:
+
+.. code:: python
+    >>> from ansys.mapdl.core import launch_mapdl
+    >>> mapdl = launch_mapdl()
+    >>> mapdl._log.info('This is an useful message')
+
+    | Level    | Instance        | Module           | Function             | Message
+    |----------|-----------------|------------------|----------------------|--------------------------------------------------------
+    | INFO     | 127.0.0.1:50052 |  test            | <module>             | This is an useful message
+
+
+
+### Other loggers
+You can create your own loggers using python ``logging`` library as you would do in any other script.
+There shall no be conflicts between these loggers.
+
+
+## Notes
+
+
+## To-Do's
+
+* [] Make sure the logging level is changed when instance is created.
+
+"""
+
+import logging
+from datetime import datetime
+import sys
+from copy import copy
+
+## Default configuration
+LOG_LEVEL = logging.DEBUG
+FILE_NAME = 'pyansys.log'
+
+# For convenience
+DEBUG = logging.DEBUG
+INFO = logging.INFO
+WARN = logging.WARN
+ERROR = logging.ERROR
+CRITICAL = logging.CRITICAL
+
+## Formatting
+
+STDOUT_MSG_FORMAT = '| %(levelname)-8s | %(instance_name)-15s |  %(module)-15s | %(funcName)-20s | %(message)s'
+FILE_MSG_FORMAT = STDOUT_MSG_FORMAT
+
+DEFAULT_STDOUT_HEADER = """
+| Level    | Instance        | Module           | Function             | Message
+|----------|-----------------|------------------|----------------------|--------------------------------------------------------
+"""
+DEFAULT_FILE_HEADER = DEFAULT_STDOUT_HEADER
+
+NEW_SESSION_HEADER = f"""\n=====================================================================================================================================
+========================                      NEW SESSION - {datetime.now().strftime("%m/%d/%Y, %H:%M:%S")}                             ========================
+====================================================================================================================================="""
```

Comment:
Keep lines to 79 characters as much as possible:
```suggestion
NEW_SESSION_HEADER = f"""
===============================================================================
       NEW SESSION - {datetime.now().strftime("%m/%d/%Y, %H:%M:%S")}
===============================================================================
"""
```